Silicon/CommonSocPkg/Library/IgdOpRegionLib/IgdOpRegionLib.c

```diff
@@ -138,19 +177,19 @@ UpdateVbt (
   if (VbtPtr != NULL) {
     DEBUG ((DEBUG_INFO, "VBT data found '%.20a'\n", VbtPtr->VbtHeader.Product_String));
     DEBUG((DEBUG_INFO, "Updating GOP VBT fields using GOP Configuration Lib.\n"));
+    VbtGen2Info = (VBT_GENERAL2_INFO *) LocateVbtBlockById ((UINT8 *)VbtPtr, VBT_BLK_ID_GENERAL2_INFO);
+    if (VbtGen2Info != NULL) {
+      for (Index = 0; Index < ChildStruct_MAX; Index++) {
+        ChildStructPtr[Index] = (CHILD_STRUCT *)(((UINT8 *)VbtGen2Info->Child_Struct) + Index * VbtPtr->VbtGen2Info.SizeChild_Struct);
```

Comment:
"VbtPtr->VbtGen2Info.SizeChild_Struct" should be changed to "VbtGen2Info ->SizeChild_Struct".
Basically we need try to avoid to use VbtPtr to direct access its member if possible since its member structure size might be changed.

Reply:
Yup. good catch!  Let us update it.